toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Strings.jsm

```diff
@@ -0,0 +1,38 @@
+
+
+class FfiConverterString {
+    static lift(buf) {
+        return this.read(new ArrayBufferDataStream(buf));
+    }
+    static lower(value) {
+        const buffer = new ArrayBuffer(this.computeSize(value));
+        const dataStream = new ArrayBufferDataStream(buffer);
+        this.write(dataStream, value)
+        return buffer;
+    }
+
+    static write(dataStream, value) {
+        const encoder = new TextEncoder();
+        const utf8Arr = encoder.encode(value);
+        dataStream.writeUint32(utf8Arr.length);
+        for (const byte of utf8Arr) {
+            dataStream.writeUint8(byte);
+        }
```

Comment:
I think there's a way to use `encoder.encodeInto()` to avoid this loop.  It's not 100% critical though.

Reply:
I avoided the loop in the `lift`/`lower` but kept it internally in the `read`/`write` to write into the `dataStream` There is a way probably of getting a slice of the dataView, converting it into a writable buffer then a Uint8Array and using `encodeInto` but for now this works, I'll keep it mind as more types are implemented

Reply:
Sounds good.  I was thinking about this case when I was initially working on it and I think I have a way to do it.  I'll open a PR once you merge.